pkg/oc/lib/newapp/cmd/newapp.go

```diff
@@ -1180,6 +1184,50 @@ func (c *AppConfig) followRefToDockerImage(ref *kapi.ObjectReference, isContext
 	return c.followRefToDockerImage(target, isContext, objects)
 }
 
+// removeRedundantTags cycles through the supplied list of objects and removes
+// any tags that are already being created in an imagestream
+func (c *AppConfig) removeRedundantTags(objects app.Objects) app.Objects {
+	objectsToRemove := map[string]struct{}{}
+	for _, obj := range objects {
+		if ist, ok := obj.(*imageapi.ImageStreamTag); ok {
+			istNamespace := ist.Namespace
+			if len(istNamespace) == 0 {
+				istNamespace = c.OriginNamespace
+			}
+			streamName, tagName, ok := imageapi.SplitImageStreamTag(ist.Name)
+			if ok {
+				is := c.findImageStreamInObjectList(objects, streamName, istNamespace)
+				if is != nil {
+					for name := range is.Spec.Tags {
+						if name == tagName {
+							objectsToRemove[fmt.Sprintf("%s/%s", istNamespace, ist.Name)] = struct{}{}
+						}
+					}
+				}
+			}
+
+		}
+	}
+
+	objectsToKeep := app.Objects{}
+	for _, obj := range objects {
+		if ist, ok := obj.(*imageapi.ImageStreamTag); ok {
+			istNamespace := ist.Namespace
+			if len(istNamespace) == 0 {
+				istNamespace = c.OriginNamespace
```

Comment:
i guess this is why/how you got rid of the suggestnamespace logic.

Reply:
correct.